### PR TITLE
qa/tasks: check if fs mounted in umount_wait

### PR DESCRIPTION
**Pull request by joscollin**
Fixes: https://tracker.ceph.com/issues/42020

### CODE REVIEW

File: qa/tasks/cephfs/fuse_mount.py [hunk 1]
```diff
@@ -305,6 +305,8 @@ def umount_wait(self, force=False, require_clean=False, timeout=900):
         """
         :param force: Complete cleanly even if the MDS is offline
         """
+        assert self.is_mounted() and self.fuse_daemon
```
Comment (batrick) [inline qa/tasks/cephfs/fuse_mount.py]: Won't this just raise an assertion that kills the DaemonWatchdog too? We probably just want to return early if it's not mounted or raise a specific NotMounted exception the DaemonWatchdog can catch.

Comment (joscollin) [reply qa/tasks/cephfs/fuse_mount.py]: Let me check more on that. I see different options to deal with this situation. I see the [kernel mount](https://github.com/ceph/ceph/blob/f90e3e102659c7930222a213e86f19d5caf736cd/qa/tasks/cephfs/kernel_mount.py#L137) does a return.

Comment (joscollin) [reply qa/tasks/cephfs/fuse_mount.py]: @batrick Fixed. Please check.

File: qa/tasks/cephfs/fuse_mount.py [hunk 2]
```diff
@@ -305,6 +305,9 @@ def umount_wait(self, force=False, require_clean=False, timeout=900):
         """
         :param force: Complete cleanly even if the MDS is offline
         """
+        if not (self.is_mounted() and self.fuse_daemon):
+            raise Exception("Filesystem not mounted")
```
Comment (batrick) [inline qa/tasks/cephfs/fuse_mount.py]: `RuntimeError` would be better.

Comment (joscollin) [reply qa/tasks/cephfs/fuse_mount.py]: Fixed

**Comment by batrick** [comment]
/ceph/teuthology-archive/pdonnell-2019-09-30_10:35:51-fs-wip-pdonnell-testing-20190930.063158-distro-basic-smithi/4348002/teuthology.log

Probably best we not throw an exception. Just return.

**Comment by joscollin** [comment]
jenkins test make check

File: qa/tasks/cephfs/fuse_mount.py [hunk 3]
```diff
@@ -305,6 +305,9 @@ def umount_wait(self, force=False, require_clean=False, timeout=900):
         """
         :param force: Complete cleanly even if the MDS is offline
         """
+        if not (self.is_mounted() and self.fuse_daemon):
```
Comment (batrick) [inline qa/tasks/cephfs/fuse_mount.py]: Jos, let's add a debug message that the mount is not unmounted. That makes it easier to confirm the PR fixes the problem and leaves a notice to future debuggers.

Comment (joscollin) [reply qa/tasks/cephfs/fuse_mount.py]: added a message

**Comment by batrick** [comment]
Was waiting for a qa run that would have otherwise failed, here it is: /ceph/teuthology-archive/pdonnell-2019-11-13_04:34:28-multimds-wip-pdonnell-testing-20191112.231923-distro-basic-smithi/4502935/teuthology.log

**Pull request merged**